src/dotnet-scaffolding/dotnet-scaffold-aspnet/Helpers/RazorPagesHelper.cs

```diff
@@ -0,0 +1,117 @@
+    internal const string DetailsModelTemplate = "DetailsModel.tt";
+    internal const string EditTemplate = "Edit.tt";
+    internal const string EditModelTemplate = "EditModel.tt";
+    internal const string IndexTemplate = "Index.tt";
+    internal const string IndexModelTemplate = "IndexModel.tt";
+    internal static Type? GetTemplateType(string? templatePath)
+    {
+        if (string.IsNullOrEmpty(templatePath))
+        {
+            return null;
+        }
+
+        Type? templateType = null;
+        switch (Path.GetFileName(templatePath))
+        {
+            case CreateTemplate:
+                templateType = typeof(Templates.RazorPages.Create);
+                break;
+            case CreateModelTemplate:
+                templateType = typeof(Templates.RazorPages.CreateModel);
+                break;
+            case IndexTemplate:
+                templateType = typeof(Templates.RazorPages.Index);
+                break;
+            case IndexModelTemplate:
+                templateType = typeof(Templates.RazorPages.IndexModel);
+                break;
+            case DeleteTemplate:
+                templateType = typeof(Templates.RazorPages.Delete);
+                break;
+            case DeleteModelTemplate:
+                templateType = typeof(Templates.RazorPages.DeleteModel);
+                break;
+            case EditTemplate:
+                templateType = typeof(Templates.RazorPages.Edit);
+                break;
+            case EditModelTemplate:
+                templateType = typeof(Templates.RazorPages.EditModel);
+                break;
+            case DetailsTemplate:
+                templateType = typeof(Templates.RazorPages.Details);
+                break;
+            case DetailsModelTemplate:
+                templateType = typeof(Templates.RazorPages.DetailsModel);
+                break;
+        }
+
+        return templateType;
+    }
+
+    internal static bool IsValidTemplate(string templateType, string templateFileName)
+    {
+        if (string.Equals(templateType, "CRUD", StringComparison.OrdinalIgnoreCase))
+        {
+            return true;
+        }
+
+        return
+            string.Equals(templateFileName, templateType, StringComparison.OrdinalIgnoreCase) ||
+            string.Equals(templateFileName, $"{templateType}Model", StringComparison.OrdinalIgnoreCase);
+    }
+
+    internal static string GetBaseOutputPath(string modelName, string? projectPath)
+    {
+        string projectBasePath = Path.GetDirectoryName(projectPath) ?? Directory.GetCurrentDirectory();
+        return Path.Combine(projectBasePath, "Pages", $"{modelName}Pages");
+    }
+
+    internal static IEnumerable<TextTemplatingProperty> GetTextTemplatingProperties(IEnumerable<string> allT4TemplatePaths, RazorPageModel razorPagesModel)
+    {
+        var textTemplatingProperties = new List<TextTemplatingProperty>();
+        foreach (var templatePath in allT4TemplatePaths)
```

Comment:
if `allT4TemplatePaths` is null , this will throw

Reply:
will add a null check